packages/flutter_tools/gradle/src/main/kotlin/dependency_version_checker.gradle.kts

```
@@ -359,3 +364,9 @@ class Version(val major: Int, val minor: Int, val patch: Int) : Comparable<Versi
         return major.toString() + "." + minor.toString() + "." + patch.toString()
     }
 }
+
+// Custom error for when the dependency_version_checker.kts script finds a dependency out of
+// the defined support range.
+class DependencyValidationException(message: String? = null, cause: Throwable? = null) : Exception(message, cause) {
+    constructor(cause: Throwable) : this(null, cause)
```

Why isn't the line like the code below? 
```suggestion
    constructor(cause: Throwable) : this(message, cause)
```

---

Good catch, it absolutely should be.

---

Oh wait sorry I confused myself - this is the constructor for the case where there is no message (single arg `cause`), and in that case we are calling our own two arg constructor `this(null, cause)` with the message as null.